build.gradle.kts

```diff
@@ -6,6 +6,7 @@ plugins {
   id("org.hypertrace.ci-utils-plugin") version "0.4.0"
   id("org.hypertrace.publish-plugin") version "1.1.1" apply false
   id("org.hypertrace.code-style-plugin") version "2.1.2" apply false
+  id("org.hypertrace.java-convention") version "0.4.0"
```

Comment:
Our java convention plugin decouples the build from the current java version. By default, it uses java 21 to compile but targets compatibility with 11. The current issue is that it's using whatever version was used to run gradle, and CI's default version was updated to 17 by GH.